pkg/operator/ceph/cluster/nodedaemon/reconcile.go

```diff
@@ -195,6 +195,28 @@ func (r *ReconcileNode) reconcile(request reconcile.Request) (reconcile.Result,
 			if err != nil {
 				return reconcile.Result{}, errors.Wrapf(err, "failed to list and delete deployments in namespace %q on node %q", namespace, request.Name)
 			}
+			err = k8sutil.DeleteServiceMonitor(r.opManagerContext, cephCluster.Namespace, cephExporterAppName)
+			if err != nil {
+				return reconcile.Result{}, errors.Wrapf(err, "failed to delete service monitor for ceph exporter in namespace %q on node %q", namespace, request.Name)
+			}
+			err = k8sutil.DeleteService(r.opManagerContext, r.context.Clientset, r.opConfig.OperatorNamespace, exporterServiceMetricName)
+			if err != nil {
+				return reconcile.Result{}, errors.Wrapf(err, "failed to delete ceph exporter metrics service in namespace %q on node %q", namespace, request.Name)
+			}
+		}
+		// Cleanup exporter if the ceph version isn't supported
+		if !cephVersion.IsAtLeast(MinVersionForCephExporter) {
+			if err := r.listDeploymentAndDelete(cephExporterAppName, request.Name, namespace); err != nil {
```

Comment:
This looks like it would just remove it from a single node? The `request.Name` is the node name, right? All ceph exporter daemons will need to be deleted, not just for a single node.